sdk/client/client.go

```diff
@@ -394,27 +394,23 @@ func (c *Client) Execute(ctx context.Context, req *Request) (*Response, error) {
 		}
 	}
 
-	// Extract OData from response
-	var o *odata.OData
-	resp.OData, err = odata.FromResponse(resp.Response)
-	if err != nil {
-		return resp, err
-	}
-	if resp == nil {
-		return resp, fmt.Errorf("nil response received")
-	}
+	// Extract OData from response, intentionally ignoring any errors as it's not crucial to extract
+	// valid OData at this point (valid json can still error here, such as any non-object literal)
+	resp.OData, _ = odata.FromResponse(resp.Response)
```

Comment:
can we add a unit test to cover #381?

Reply:
sounds reasonable to me, I don't think it blocks this PR - but we should look to make that more testable/add a PR to add a test for that in the future 👍